courseraprogramming/commands/publish.py

```diff
@@ -0,0 +1,223 @@
+                course_id,
+                item_id,
+                metadata)
+            write_std_out("Publish complete for item {} in course {}\n".format(
+                item_id, course_id), args)
+        except ItemNotFoundError as e:
+            logging.error(
+                "Unable to find a publishable assignment with item "
+                "id {}. Maybe there are no changes to publish?".format(
+                    item_id, course_id))
+            sys.exit(ErrorCodes.FATAL_ERROR)
+        except ValidationError as e:
+            logging.error(
+                "We found some validation errors in your assignment with item "
+                "id {}. Please verify that your assignment is formatted "
+                "correctly and try again.".format(
+                    item_id))
+            sys.exit(ErrorCodes.FATAL_ERROR)
+        except GraderExecutorError as e:
+            if e.status == GraderExecutorStatus.PENDING:
+                logging.warn(
+                    "We are still processing your grader for your assignment "
+                    "with item id {}.  Please try again soon.".format(
+                        item_id))
+                sys.exit(ErrorCodes.RETRYABLE_ERROR)
+            elif e.status == GraderExecutorStatus.FAILED:
+                logging.error(
+                    "We were unable to process your grader for your "
+                    "assignment with item id {}.  Please try to upload your "
+                    "grader again. If the problem persists, please let us "
+                    "know.".format(
+                        item_id))
+                sys.exit(ErrorCodes.FATAL_ERROR)
+            elif e.status == GraderExecutorStatus.MISSING:
+                logging.error(
+                    "We were unable to find your grader for your assignment "
+                    "with item id {}.  Please try to upload your grader "
+                    "again. If the problem persists, please let us "
+                    "know.".format(
+                        item_id))
+                sys.exit(ErrorCodes.FATAL_ERROR)
+        except InternalError as e:
+            logging.error(
+                "Something unexpected happened while trying to publish your "
+                "assignment with item id {}. Please verify your course and "
+                "item ids are correct.  If the problem persists, please let "
+                "us know.".format(
+                    item_id))
+            sys.exit(ErrorCodes.FATAL_ERROR)
+
+
+def get_metadata(oauth2_instance, get_endpoint, course_id, item_id):
+    auth = oauth2_instance.build_authorizer()
+    resp = requests.get(
+        '{}/{}~{}'.format(get_endpoint, course_id, item_id),
+        auth=auth)
+    if resp.status_code == 404:
+        raise ItemNotFoundError(course_id, item_id)
+    elif resp.status_code == 500:
+        raise InternalError()
+    return resp.json()['elements'][0]['metadata']
+
+
+def post_publish(
```

Comment:
`post_publish` is not the most descriptive name. Could you either use a better name, or add a docstring to describe what the purpose of this is?